loopchain/channel/channel_inner_service.py

```diff
@@ -838,71 +837,56 @@ def get_invoke_result(self, tx_hash):
             return message_code.Response.fail, None
 
     @message_queue_task
-    async def get_block_v2(self, block_height, block_hash, block_data_filter, tx_data_filter):
+    async def get_block_v2(self, block_height, block_hash) -> Tuple[int, str, str]:
         # This is a temporary function for v2 support of exchanges.
-        block, block_filter, block_hash, _, fail_response_code, tx_filter = \
-            await self.__get_block(block_data_filter, block_hash, block_height, tx_data_filter)
+        block, block_hash, _, fail_response_code = await self.__get_block(block_hash, block_height)
         if fail_response_code:
-            return fail_response_code, block_hash, json.dumps({}), ""
+            return fail_response_code, block_hash, json.dumps({})
 
         tx_versioner = self._blockchain.tx_versioner
         bs = BlockSerializer.new(block.header.version, tx_versioner)
         block_data_dict = bs.serialize(block)
 
         if block.header.height == 0:
-            return message_code.Response.success, block_hash, json.dumps(block_data_dict), []
+            return message_code.Response.success, block_hash, json.dumps(block_data_dict)
 
-        confirmed_tx_list = block_data_dict["confirmed_transaction_list"]
         confirmed_tx_list_without_fail = []
-
-        tss = {
-            "genesis": TransactionSerializer.new("genesis", tx_versioner),
-            "0x2": TransactionSerializer.new("0x2", tx_versioner),
-            "0x3": TransactionSerializer.new("0x3", tx_versioner)
-        }
-
-        for tx in confirmed_tx_list:
-            version = tx_versioner.get_version(tx)
-            tx_hash = tss[version].get_hash(tx)
-
-            invoke_result = self._block_manager.get_invoke_result(tx_hash)
+        for tx in block.body.transactions.values():
+            invoke_result = self._block_manager.get_invoke_result(tx.hash)
 
             if 'failure' in invoke_result:
                 continue
 
-            if tx_versioner.get_version(tx) == "0x3":
+            if tx.version == "0x3":
                 step_used, step_price = int(invoke_result["stepUsed"], 16), int(invoke_result["stepPrice"], 16)
-                tx["fee"] = hex(step_used * step_price)
+                tx.raw_data["fee"] = hex(step_used * step_price)
```

Comment:
Modifying `raw_data` directly is not recommended.
```python
ts = TransactionSerializer.new(tx.version, tx.type(), tx_versioner)
data = ts.to_full_data(tx) 
data["fee"] = hex(step_used * step_price)
data.pop("stepUsed")
data.pop("stepPrice")
```

Note that `to_full_data` has additional attributes.
v2 tx
- "method": "icx_sendTransaction"

v3 tx
- "txHash": "0xabcd1234"

raw data may not have "txHash" of v3 tx

Reply:
Thx for great suggestion!